lib/src/adder.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// adder.dart
+// Implementation of Adder Module.
+//
+// 2023 June 1
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// An abstract class for all adder module.
+abstract class Adder extends Module {
+  /// The input to the adder pin [a].
+  Logic a;
+
+  /// The input to the adder pin [b].
+  Logic b;
+
+  /// The addition results [sum].
+  Logic get sum;
+
+  /// Takes in input [a] and input [b] and return the [sum] of the addition
+  /// result.
+  Adder(this.a, this.b, {super.name});
```

Comment:
You could take care of the `addInput`s, width checks, etc. in here since they should be common across all adder implementations.

same applies to the multiplier